src/Raven.Server/Documents/AttachmentsStorage.Sharding.cs

```diff
@@ -42,14 +41,19 @@ internal static ByteStringContext.Scope GenerateBucketAndEtagIndexKeyForAttachme
                 (int)AttachmentsTable.Etag, ref tvr, out slice);
         }
 
+        internal static void UpdateBucketStatsForAttachments(Transaction tx, Slice key, ref TableValueReader oldValue, ref TableValueReader newValue)
+        {
+            ShardedDocumentsStorage.UpdateBucketStatsInternal(tx, key, ref newValue, changeVectorIndex: (int)AttachmentsTable.ChangeVector, sizeChange: newValue.Size - oldValue.Size);
+        }
+
         private void UpdateBucketStatsOnPutOrDeleteStream(DocumentsOperationContext context, Slice attachmentKey, long sizeChange)
         {
             if (_documentDatabase is not ShardedDocumentDatabase)
                 return;
 
             using (GetBucketFromAttachmentKey(context, attachmentKey, out var bucketSlice))
             {
-                ShardedDocumentsStorage.UpdateBucketStats(context.Transaction.InnerTransaction, bucketSlice, oldSize: 0, newSize: sizeChange);
+                ShardedDocumentsStorage.UpdateBucketStatsInternal(context.Transaction.InnerTransaction, key: bucketSlice, value: ref TableValueReaderUtils.EmptyReader, changeVectorIndex: -1, sizeChange: sizeChange);
```

Comment:
not a fan of the `-1` & `TableValueReaderUtils.EmptyReader` would it possible to have maybe some other overload?

Reply:
done